src/foam/core/CompoundException.java

```diff
@@ -7,19 +7,31 @@
 package foam.core;
 
 import java.util.ArrayList;
+import foam.nanos.app.AppConfig;
+import foam.nanos.app.Mode;
 
 public class CompoundException
   extends RuntimeException
 {
-  private ArrayList exceptions_ = new ArrayList<RuntimeException>();
+  private ArrayList exceptions_ = new ArrayList<Throwable>();
 
   public void add(Throwable t) {
     exceptions_.add(t);
   }
 
-  public void maybeThrow() {
-    if ( exceptions_.size() != 0 ) {
-      throw this;
+  public String toString() {
+    return "CompoundException [exceptions_=" + exceptions_ + "]";
+  }
+
+  public String getMessage() {
+    return super.getMessage();
```

Comment:
How is this useful?

Reply:
It should create a StringBuilder and concatenate the getMessage() of each of the exception_s.